chromium_src/services/network/cookie_expiry_browsertest.cc

```diff
@@ -0,0 +1,179 @@
+  CHECK(rv);
+}
+
+void JSCookieStoreWriteCookie(Browser* browser, std::string expires_in_ms) {
+  content::EvalJsResult result = content::EvalJs(
+      browser->tab_strip_model()->GetActiveWebContents(),
+      base::StringPrintf("async function doSet() {"
+                         "  await window.cookieStore.set("
+                         "       { name: 'name',"
+                         "         value: 'Good',"
+                         "         expires: Date.now() + %s,"
+                         "          });"
+                         "  window.domAutomationController.send(true);"
+                         "}"
+                         "doSet()",
+                         expires_in_ms.c_str()),
+      content::EXECUTE_SCRIPT_USE_MANUAL_REPLY);
+}
+
+void SetCookieDirect(Browser* browser,
+                     const GURL& url,
+                     const std::string& cookie_line) {
+  net::CookieOptions options;
+  options.set_include_httponly();
+  options.set_same_site_cookie_context(
+      net::CookieOptions::SameSiteCookieContext::MakeInclusive());
+
+  auto cookie_obj = net::CanonicalCookie::Create(
+      url, cookie_line, base::Time::Now(), absl::nullopt /* server_time */,
+      absl::nullopt /* cookie_partition_key */);
+
+  base::RunLoop run_loop;
+  browser->tab_strip_model()
+      ->GetActiveWebContents()
+      ->GetBrowserContext()
+      ->GetDefaultStoragePartition()
+      ->GetCookieManagerForBrowserProcess()
+      ->SetCanonicalCookie(
+          *cookie_obj, url, options,
+          base::BindLambdaForTesting(
+              [&](net::CookieAccessResult status) { run_loop.Quit(); }));
+  run_loop.Run();
+}
+
+std::vector<net::CanonicalCookie> GetAllCookiesDirect(Browser* browser) {
+  base::RunLoop run_loop;
+  std::vector<net::CanonicalCookie> cookies_out;
+  browser->tab_strip_model()
+      ->GetActiveWebContents()
+      ->GetBrowserContext()
+      ->GetDefaultStoragePartition()
+      ->GetCookieManagerForBrowserProcess()
+      ->GetAllCookies(base::BindLambdaForTesting(
+          [&run_loop,
+           &cookies_out](const std::vector<net::CanonicalCookie>& cookies) {
+            cookies_out = cookies;
+            run_loop.Quit();
+          }));
+  run_loop.Run();
+  return cookies_out;
+}
+
+IN_PROC_BROWSER_TEST_F(CookieExpirationTest, CheckExpirationTimeForJSCookie) {
+  GURL url = https_server_->GetURL("a.com", "/simple.html");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+  std::cout << "cookie long expiry in days "
+            << std::to_string(k4YearsInDays.InSeconds());
+  JSDocumentCookieWriteCookie(
+      browser(), "max-age=" + std::to_string(k4YearsInDays.InSeconds()));
+  std::vector<net::CanonicalCookie> all_cookies =
+      GetAllCookiesDirect(browser());
+  EXPECT_EQ(1u, all_cookies.size());
+  for (const net::CanonicalCookie& cookie : all_cookies) {
+    EXPECT_EQ((cookie.ExpiryDate().ToDeltaSinceWindowsEpoch().InDays() -
```

Comment:
is `ToDeltaSinceWindowsEpoch()` necessary? `Time - Time` should return `TimeDelta`.